docs/developer-guide/working-with-rules.md

```diff
@@ -311,6 +312,117 @@ Keep in mind that comments are technically not a part of the AST and are only at
 
 **Note:** One of the libraries adds AST node properties for comments - do not use these properties. Always use `sourceCode.getComments()` as this is the only guaranteed API for accessing comments (we will likely change how comments are handled later).
 
+### Accessing Code Paths
+
+ESLint is analyzing code paths while traversing AST. You can access that code path objects with `context.getCodePath()`.
+`context.getCodePath()` returns a `CodePath` object of the current location.
+
+> **Note:** That code path has not completed. If you want to get the completed code path, you can use events which will be described later.
```

Comment:
This seems like it could potentially cause a lot of confusion. If the `CodePath` object isn't completed when someone calls `getCodePath()`, that could lead to errors, correct?

If yes, is there a way to do the analysis first? I'm a little worried that we could see a lot of errors relating to `getCodePath()` if people expect the object to be complete (like `getScope()`).